plugins/opa/examples/containerd-hostpath-cve-2021-43816/policy.rego

```diff
@@ -0,0 +1,66 @@
+package fairwinds
+
+podHasHostPathVolume(pod) {
+        # Get the volumes from each container
+        volumes := pod.spec.volumes[_]
+        volumes["hostPath"]
+}
+
+blockMountPath(volumeMount) {
+    badPath := input.parameters.badPaths[_]
+        volumeMount["mountPath"] == badPath
+}
+
+podHasDangerousVolumeMount(pod) {
+  container := pod.spec.containers[_]
+  volumeMount := container.volumeMounts[_]
+  blockMountPath(volumeMount)
+}
+
+checkCronjob[actionItem] {
+        input.kind == "CronJob"
+        pod := input.spec.jobTemplate.spec.template
+    podHasHostPathVolume(pod)
+        podHasDangerousVolumeMount(pod)
+        actionItem := {
+                "title": concat(" ", [input.kind, " may be exploiting CVE-2021-43816"]),
+                "description": "Pod has a hostPath mount and a volumeMount at a vulnerable path. It may be exploiting CVE-2021-43816",
+                "remediation": "Immediately investigate this workload and mitigate CVE-2021-43816",
+                "category": "Security",
+        "severity": .99
+        }
+}
+
+checkDeploymentLike[actionItem] {
+        kinds := {"Deployment", "DaemonSet", "StatefulSet", "ReplicaSet", "Job"}
+        kind := kinds[_]
+        input.kind == kind
+        pod := input.spec.template
+    podHasHostPathVolume(pod)
+        podHasDangerousVolumeMount(pod)
+        actionItem := {
+                "title": concat(" ", [input.kind, " may be exploiting CVE-2021-43816"]),
+                "description": "Pod has a hostPath mount and a volumeMount at a vulnerable path. It may be exploiting CVE-2021-43816",
+                "remediation": "Immediately investigate this workload and mitigate CVE-2021-43816",
+                "category": "Security",
+        "severity": .99
+
+        }
+}
+
+checkPod[actionItem] {
+        input.kind == "Pod"
+
+        # Only alert for stand-alone pods,
+        # avoiding duplicate action-items for pods which belong to a controller.
+        not input.metadata.ownerReferences
+    podHasHostPathVolume(input)
+        podHasDangerousVolumeMount(input)
+        actionItem := {
```

Comment:
I'm sure there's a way to extract out the common stuff once you have the pod spec. But I don't really know what it is off hand.